NPE: While showing customer details

**Summary:** 

I was viewing the customer with identifier 123 and App crashed 

Here is log 

```
Process: com.mifos.apache.fineract, PID: 30693
java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.
at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:111)
at android.os.Handler.handleCallback(Handler.java:746)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:148)
at android.app.ActivityThread.main(ActivityThread.java:5443)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
Caused by: java.lang.IndexOutOfBoundsException: Invalid index 1, size is 1
at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
at java.util.ArrayList.get(ArrayList.java:308)
at com.mifos.apache.fineract.ui.online.customerdetails.CustomerDetailsFragment.showCustomerDetails(CustomerDetailsFragment.java:206)
at com.mifos.apache.fineract.ui.online.customerdetails.CustomerDetailsPresenter$1.onNext(CustomerDetailsPresenter.java:59)
at com.mifos.apache.fineract.ui.online.customerdetails.CustomerDetailsPresenter$1.onNext(CustomerDetailsPresenter.java:55)
at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:198)
at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:250)
at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:109)
at android.os.Handler.handleCallback(Handler.java:746) 
at android.os.Handler.dispatchMessage(Handler.java:95) 
at android.os.Looper.loop(Looper.java:148) 
at android.app.ActivityThread.main(ActivityThread.java:5443) 
at java.lang.reflect.Method.invoke(Native Method) 
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618) 
```

--- a/app/src/main/java/com/mifos/apache/fineract/data/models/customer/ContactDetail.java
+++ b/app/src/main/java/com/mifos/apache/fineract/data/models/customer/ContactDetail.java
@@ -34,8 +34,8 @@ public final class ContactDetail {
         super();
     }
 
-    public String getType() {
-        return this.type.name();
+    public Type getType() {
+        return this.type;
     }
 
     public void setType(final Type type) {

--- a/app/src/main/java/com/mifos/apache/fineract/ui/online/customerdetails/CustomerDetailsContract.java
+++ b/app/src/main/java/com/mifos/apache/fineract/ui/online/customerdetails/CustomerDetailsContract.java
@@ -1,5 +1,6 @@
 package com.mifos.apache.fineract.ui.online.customerdetails;
 
+import com.mifos.apache.fineract.data.models.customer.ContactDetail;
 import com.mifos.apache.fineract.data.models.customer.Customer;
 import com.mifos.apache.fineract.ui.base.MvpView;
 
@@ -14,6 +15,8 @@ public interface CustomerDetailsContract {
         void showUserInterface();
 
         void showCustomerDetails(Customer customer);
+
+        void showContactDetails(ContactDetail contactDetail);
 
         void showToolbarTitleSubtitle(String title, String subtitle);
 

--- a/app/src/main/java/com/mifos/apache/fineract/ui/online/customerdetails/CustomerDetailsFragment.java
+++ b/app/src/main/java/com/mifos/apache/fineract/ui/online/customerdetails/CustomerDetailsFragment.java
@@ -16,6 +16,7 @@ import android.widget.TextView;
 
 import com.mifos.apache.fineract.R;
 import com.mifos.apache.fineract.data.models.customer.Address;
+import com.mifos.apache.fineract.data.models.customer.ContactDetail;
 import com.mifos.apache.fineract.data.models.customer.Customer;
 import com.mifos.apache.fineract.ui.base.MifosBaseActivity;
 import com.mifos.apache.fineract.ui.base.MifosBaseFragment;
@@ -202,9 +203,9 @@ public class CustomerDetailsFragment extends MifosBaseFragment
             tvPhoneNo.setVisibility(View.GONE);
             tvMobileNo.setVisibility(View.GONE);
         } else {
-            tvEmail.setText(customer.getContactDetails().get(0).getValue());
-            tvPhoneNo.setText(customer.getContactDetails().get(1).getValue());
-            tvMobileNo.setText(customer.getContactDetails().get(2).getValue());
+            for (ContactDetail contactDetail : customer.getContactDetails()) {
+                showContactDetails(contactDetail);
+            }
         }
 
         tvBirthDay.setText(customer.getDateOfBirth().getYear() + "-" +
@@ -219,6 +220,24 @@ public class CustomerDetailsFragment extends MifosBaseFragment
             subtitle = getString(R.string.assigned_employee) + " " + customer.getAssignedEmployee();
         }
         showToolbarTitleSubtitle(title, subtitle);
+    }
+
+    @Override
+    public void showContactDetails(ContactDetail contactDetail) {
+        switch (contactDetail.getType()) {
+            case EMAIL:
+                tvEmail.setText(contactDetail.getValue());
+                tvEmail.setVisibility(View.VISIBLE);
+                break;
+            case MOBILE:
+                tvMobileNo.setText(contactDetail.getValue());
+                tvMobileNo.setVisibility(View.VISIBLE);
+                break;
+            case PHONE:
+                tvPhoneNo.setText(contactDetail.getValue());
+                tvPhoneNo.setVisibility(View.VISIBLE);
+                break;
+        }
     }
 
     @Override

--- a/app/src/main/res/layout/fragment_customer_details.xml
+++ b/app/src/main/res/layout/fragment_customer_details.xml
@@ -171,27 +171,30 @@
                             android:layout_width="wrap_content"
                             android:paddingBottom="@dimen/layout_padding_4dp"
                             android:paddingTop="@dimen/layout_padding_4dp"
+                            android:visibility="gone"
                             tools:text="example@gmail.com"/>
 
                         <TextView
-                            android:drawableLeft="@drawable/ic_smartphone_black_24dp"
-                            android:drawablePadding="@dimen/layout_padding_4dp"
-                            android:drawableStart="@drawable/ic_smartphone_black_24dp"
+                            android:drawableStart="@drawable/ic_phone_black_24dp"
+                            android:drawableLeft="@drawable/ic_phone_black_24dp"
                             android:id="@+id/tv_phone_no"
+                            android:drawablePadding="@dimen/layout_padding_4dp"
                             android:layout_height="wrap_content"
                             android:layout_width="wrap_content"
                             android:paddingBottom="@dimen/layout_padding_4dp"
                             android:paddingTop="@dimen/layout_padding_4dp"
+                            android:visibility="gone"
                             tools:text="PrefixOperator - NoTelp"/>
 
                         <TextView
-                            android:drawableLeft="@drawable/ic_phone_black_24dp"
                             android:drawablePadding="@dimen/layout_padding_4dp"
-                            android:drawableStart="@drawable/ic_phone_black_24dp"
+                            android:drawableLeft="@drawable/ic_smartphone_black_24dp"
+                            android:drawableStart="@drawable/ic_smartphone_black_24dp"
                             android:id="@+id/tv_mobile_no"
                             android:layout_height="wrap_content"
                             android:layout_width="wrap_content"
                             android:paddingTop="@dimen/layout_padding_4dp"
+                            android:visibility="gone"
                             tools:text="KodeArea - Notelp"/>
 
                     </LinearLayout>

